Updating form output for school finder modal.

weerd opened the pull request:
Fixes #5788
#### What's this PR do?

There was an additional form wrapper for elements that did not contian any markup or content but was still being output as an empty div with styled padding which was causing the extra spacing.
#### Where should the reviewer start?

Quick :eyes: on the one file changed.
#### Any background context you want to provide?

@DFurnes do you happen to know if something gets otherwise populated into the `elements` container in some other version of the school finder modal (if there is one) that I might not be accounting for?
#### What are the relevant tickets?
#5788

---

@DFurnes 

cc: @angaither

Commit:
Updating form output for school finder modal.

There was an additional form wrapper for elements that did not contian any markup or content but was still being output as an empty div with styled padding which was causing the extra spacing.

DFurnes commented:
Hmmm, it looks like the sections on the modal [were intended to be placed in there](https://github.com/DoSomething/phoenix/commit/19915b0c9302d941e7a191a76875665ad0f4d249). I don't really remember why that was the way I set things up though. :sweat_smile: 

This seems reasonable, though it might be good to remove the following lines (like [this one](https://github.com/DoSomething/phoenix/blob/510b845faef5c57eb1751622d72bac090c46fa6b/lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc#L319)) that reference `$form['elements']` since it won't exist anymore. It also looks like the t-shirt form used that container, but that's RIP :skull: so probably no worries there.

weerd commented:
Hmmm, good call. I was trying to see if there was a way to check if stuff was getting placed in there and if empty then unset it....?

DFurnes commented:
Yeah, ideally we only create the container if we need it. But it's strange that it's showing up as empty since it looks like we're putting things in there a few lines below. :confused:

weerd commented:
So the issue is that since it's building a Droooooooopal render array thing, it populates with some preliminary stuff like the `#attributes` and `#type` so there will technically _always_ be something in the array when it checks whether to output or not.

Commit:
Updating approach to handle empty [elements] array item and empty output.

Instead of removing the [elements] item entirely, which could break modal in different scenarios, we just remove the item at the end of the form generation if the item only has the default array elements and nothing else; thus it is empty.

DFurnes commented:
:grimacing: :ship: :+1:
